### PR TITLE
GG-25831 Fixed IllegalArgumentException thrown by HibernateL2CacheExa…

### PR DESCRIPTION
**Pull request by sanpwc**
…mple for ignite-hibernate_5.3

### CODE REVIEW

File: examples/src/main/java-lgpl/org/apache/ignite/examples/datagrid/hibernate/HibernateL2CacheExample.java [hunk 1]
```diff
@@ -257,4 +255,46 @@ private static void printStats(SessionFactory sesFactory) {
...
+            // Represents the name of query results region specific to hibernate 5.3 {@see IgniteQueryResultsRegion}.
+            "default-query-results-region";
+    }
+
+    /**
+     * Returns {@code true} if ignite-hibernate 5.1 is enabled.
+     *
+     * @return {@code true} if ignite-hibernate 5.1 is enabled.
+     */
+    private static boolean isIgniteHibernate51Enabled() {
```
Comment (sanpwc) [inline examples/src/main/java-lgpl/org/apache/ignite/examples/datagrid/hibernate/HibernateL2CacheExample.java]: The method name seems a bit inaccurate. HibernateTimestampsRegion is also available in other hibernate versions besides 5.1 (for example 4.2).

File: examples/src/main/java-lgpl/org/apache/ignite/examples/datagrid/hibernate/HibernateL2CacheExample.java [hunk 2]
```diff
@@ -257,4 +255,46 @@ private static void printStats(SessionFactory sesFactory) {
...
         System.out.println("=====================================");
     }
+
+    /**
+     * Returns the name of the timestamps cache to a specific version of apache-hibernate.
+     *
+     * @return Name of the update timestamps cache.
+     */
+    private static String timestampsCacheName() {
+        return isIgniteHibernate51Enabled()?
```
Comment (sanpwc) [inline examples/src/main/java-lgpl/org/apache/ignite/examples/datagrid/hibernate/HibernateL2CacheExample.java]: Space is missing.

File: examples/src/main/java-lgpl/org/apache/ignite/examples/datagrid/hibernate/HibernateL2CacheExample.java [hunk 3]
```diff
@@ -257,4 +255,46 @@ private static void printStats(SessionFactory sesFactory) {
...
+            "default-update-timestamps-region";
+    }
+
+    /**
+     * Returns the name of the query results cache to a specific version of apache-hibernate.
+     *
+     * @return Name of the update timestamps cache.
+     */
+    private static String queryResultsCacheName() {
+        return isIgniteHibernate51Enabled()?
```
Comment (sanpwc) [inline examples/src/main/java-lgpl/org/apache/ignite/examples/datagrid/hibernate/HibernateL2CacheExample.java]: Space is missing.

File: examples/src/main/java-lgpl/org/apache/ignite/examples/datagrid/hibernate/HibernateL2CacheExample.java (hunk 2 above)
Comment (sk0x50) [reply examples/src/main/java-lgpl/org/apache/ignite/examples/datagrid/hibernate/HibernateL2CacheExample.java]: Thanks! Will fix it.

File: examples/src/main/java-lgpl/org/apache/ignite/examples/datagrid/hibernate/HibernateL2CacheExample.java (hunk 1 above)
Comment (sk0x50) [reply examples/src/main/java-lgpl/org/apache/ignite/examples/datagrid/hibernate/HibernateL2CacheExample.java]: To be honest, this example does not even compile for ignite-hibernate4.2. :)
Anyway, I would propose `isIgniteHibernate52Enabled`. What do you think?

**Pull request closed without merging**